codec-socks/src/main/java/io/netty/handler/codec/socksx/v5/Socks5AddressDecoder.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.handler.codec.socksx.v5;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.handler.codec.DecoderException;
+import io.netty.util.CharsetUtil;
+import io.netty.util.NetUtil;
+
+/**
+ * Decodes a SOCKS5 address field into its string representation.
+ *
+ * @see Socks5CommandRequestDecoder
+ * @see Socks5CommandResponseDecoder
+ */
+public interface Socks5AddressDecoder {
+
+    Socks5AddressDecoder DEFAULT = new Socks5AddressDecoder() {
+        @Override
+        public String decodeAddress(Socks5AddressType addrType, ByteBuf in) throws Exception {
+            if (addrType == Socks5AddressType.IPv4) {
+                return NetUtil.intToIpAddress(in.readInt());
+            }
+            if (addrType == Socks5AddressType.DOMAIN) {
+                return in.readSlice(in.readUnsignedByte()).toString(CharsetUtil.US_ASCII);
+            }
+            if (addrType == Socks5AddressType.IPv6) {
+                byte[] tmp = new byte[16];
+                in.readBytes(tmp);
+                return NetUtil.bytesToIpAddress(tmp, 0, tmp.length);
+            } else {
+                throw new DecoderException("unsupported address type: " + (addrType.byteValue() & 0xFF));
+            }
+        }
+    };
+
+    /**
+     * Decodes a SOCKS5 address field into its string representation.
+     *
+     * @param addrType the type of the address
+     * @param in the input buffer which contains the SOCKS5 address field at its reader index
+     */
+    String decodeAddress(Socks5AddressType addrType, ByteBuf in) throws Exception;
```

Comment:
I wonder if we need to make clear that the user not need to check for enough bytes in the buffer ?